### ISSUE TITLE
untyped can return a TypedActorRef

### ISSUE THREAD

**Issue by quartusdev** [Horusiath/Akkling#59]
Executing the following piece of code results in a maybe unexpected output:
```
    let x = spawnAnonymous system <| props (actorOf (fun (msg: obj) -> Unhandled))
    let (y: IActorRef<string>) = typed x
    let z = untyped y

    // Print types
    printf "%s %A\n" <| x.GetType().Name <| x.GetType().GetGenericArguments()
    printf "%s %A\n" <| y.GetType().Name <| y.GetType().GetGenericArguments()
    printf "%s %A\n" <| z.GetType().Name <| z.GetType().GetGenericArguments()
```
The output is:
```
TypedActorRef`1 [|System.Object|]
TypedActorRef`1 [|System.String|]
TypedActorRef`1 [|System.Object|]
```
I would expected ``z`` to be a "native" Akka.Actor.IActorRef, but it is not. The problem seems to be [``typed``](https://github.com/Horusiath/Akkling/blob/master/src/Akkling/ActorRefs.fs#L134). The method checks for the actor ref being a TypedActorRef of the same type, but when it is not simply wraps it in a new TypeActorRef. In the case above that means wrapping a TypedActorRef<string> in a TypedActorRef<obj>.
Because of how generics work, there is no easy solution to fixing the method without touching anything else or using reflection. So I think there should be a non generic interface that all typed actor refs implement that has atleast has one property to access the underlying IActorRef. This way ``typed`` could use that to retrieve an actor ref to wrap.

**Comment by Horusiath** [comment]
You're right. `typed` function should check if provided `IActorRef` is not typed already and in that case return new `TypedActorRef<>` with generic param updated.

**Comment by quartusdev** [comment]
Fixed with #66
